docs/madr/decisions/035-mesh-metric-policy.md

```diff
@@ -0,0 +1,126 @@
+# MeshMetric policy
+* status: accepted
+  Technical Story: https://github.com/kumahq/kuma/issues/5708
+
+## Context and Problem statement
+We want to create a new policy responsible for Metrics collection. This policy will be part of Kuma observability policies 
+stack, alongside `MeshAccessLog` and `MeshTrace`. The new policy should expose the same functionality as now is exposed through `Mesh`. 
+We want to provide support for Prometheus and native support for OTEL backend.
+
+## Decision Outcome
+We will implement new `MeshMetric` policy as described further in this MADR.
+
+## Solution
+
+### API design
+
+#### Current API
+The API is going to be similar to what we have under `mesh.spec.metrics`. The current API looks like this:
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: Mesh
+metadata:
+  name: default
+spec:
+  metrics:
+  enabledBackend: prometheus-1
+  backends:
+    - name: prometheus-1
+      type: prometheus
+      conf:
+        tls:
+          mode: disabled | providedTLS | activeMTLSBackend
+        port: 5670
+        path: /metrics
+        tags:
+          kuma.io/service: dataplane-metrics
+        envoy:
+          filterRegex: http2_act.*
+          usedOnly: true
+        aggregate:
+          - name: my-service
+            path: "/metrics/prometheus"
+            port: 8888
+          - name: other-sidecar
+            port: 8000
+```
+
+#### Changes to the API
+1. I think the aggregation that [uses the aggregate name and pod annotations](https://kuma.io/docs/2.5.x/policies/traffic-metrics/#expose-metrics-from-applications) no longer makes sense, as we can use targetRef 
+policies to point to specific DPs. I imagine there would be a default wide policy (possibly `Mesh` scoped) and per app / framework / language overrides.
+2. What I also don’t like about this is that “filters” are under envoy - I’d probably rename it to filter and introduce 
+the concept of “profiles” like we discussed. Maybe something like this:
+
+```yaml
+filter:
+  usedOnly: true # true or false
+  profile: minimal # one of minimal, default, full
```

Comment:
Is it only predefined profiles? Is a list of metrics for each profile a separate MADR?

Reply:
I think that, at least at the beginning, we should allow only predefined profiles. I will work with the envoy stats and prepare a list of metrics for the given profiles to include in this MADR

Reply:
Just realising that this is cool because you can add more stats just for the dataplane you are investigating

Reply:
I would like to push this to separate MADR, as there is too much metrics to easily create profiles out of them, and since we don't have such a possibility currently we won't need this in first version of this policy